sabnzbd/downloader.py

```diff
@@ -711,22 +716,23 @@ def run(self):
 
                 else:
                     try:
-                        article.nzf.nzo.update_download_stats(sabnzbd.BPSMeter.bps, server.id, bytes_received)
+                        article.nzf.nzo.update_download_stats(BPSMeter.bps, server.id, bytes_received)
                     except AttributeError:
                         # In case nzf has disappeared because the file was deleted before the update could happen
                         pass
 
-                    sabnzbd.BPSMeter.update(server.id, bytes_received, force_full_update=False)
+                    BPSMeter.cached_amount[server.id] += bytes_received
+                    BPSMeter.sum_cached_amount += bytes_received
```

Comment:
Not sure if I can approve of this.. We shouldn't be accessing `BPSMeter` internals outside. 
If this is not moved here but kept in the `.update()`, how much is the improvement still?
I kind of assume most of the improvement is due to the initialization of the variables beforehand.

Reply:
Hang on, I'll do some tests with only the time command to see if it's yappi that messes it up.

Reply:
```
It seems like it's the call itself that takes time, not what it does
```
Correct. We found exactly the same overhead savings by switching to attributes from function properties. Yes, the call itself takes time.

Reply:
In this case I would really like to see if it makes noticeable % difference in actual download performance. Clean code is also valuable.
Calling functions indeed is overhead, but that could be extended too far where everything is just thrown in the main Downloader loop.